projects/core/src/cart/facade/active-cart.service.ts

```diff
@@ -70,40 +80,39 @@ export class ActiveCartService implements OnDestroy {
   }
 
   protected initActiveCart() {
+    // Any change of user id is also interesting for us, because we have to merge/load/switch cart in those cases.
     this.subscription.add(
-      combineLatest([
-        this.userIdService.getUserId(),
-        this.activeCartId$.pipe(auditTime(0)),
-      ])
-        .pipe(map(([userId]) => userId))
-        .subscribe((userId) => {
-          this.userId = userId;
-          if (this.userId !== OCC_USER_ID_ANONYMOUS) {
-            if (this.isJustLoggedIn(userId)) {
-              this.loadOrMerge(this.cartId);
-            }
+      this.userIdService
+        .getUserId()
+        .pipe(
+          // We never trigger merge/load login on app initialization here and that's why we wait with pairwise for a change of userId (not initialization).
+          pairwise(),
+          switchMap(([previousUserId, userId]) =>
+            // We need cartId once we have the previous and current userId. We don't want to subscribe to cartId stream before.
+            combineLatest([
+              of(previousUserId),
+              of(userId),
+              this.activeCartId$,
+            ]).pipe(take(1))
```

Comment:
I would move this take(1) to outer pipe, for better readability, but it's very minor ;-)

Reply:
Yeah and broke the merging :P

Reply:
Fair point :-) Sorry!